carapace-server/src/main/java/org/carapaceproxy/server/mapper/requestmatcher/RequestMatcher.java

```diff
@@ -19,14 +19,18 @@
  */
 package org.carapaceproxy.server.mapper.requestmatcher;
 
-import io.netty.handler.codec.http.HttpRequest;
-
 /**
  * Generic criteria to apply a route to a request
  */
 public interface RequestMatcher {
 
-    boolean matches(HttpRequest request);
+    /**
+     * Checks request appliance to defined criteria.
+     * @param context over the request appliance is checked.
+     * @return
+     * @throws MatchingException whether there are troubles in context inspection.
+     */
+    boolean matches(MatchingContext context) throws MatchingException;
```

Comment:
I am not sure that we need a 'MatchingException'
the caller won't be able to do anything
any "internal error" should end with:
- a LOG message at level SEVERE
- return *false*